### PR TITLE
prevent support-series migration tests from failing when other migrations are added in the future.

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1204/
Test PASSed.

**Comment by urosj** [comment]
:+1: QA passed

File: internal/charmstore/migrations_test.go [hunk 1]
```diff
@@ -341,12 +344,14 @@ func getMigrations(names ...mongodoc.MigrationName) (ms []migration) {
 	return ms
 }
 
-func (s *migrationsSuite) checkEntity(c *gc.C, expectEntity *mongodoc.Entity) {
+func (s *migrationsSuite) checkEntity(c *gc.C, expectEntity *mongodoc.Entity, includeFields []string) {
```
Comment (rogpeppe) [inline internal/charmstore/migrations_test.go]: A doc comment would be good here please.

File: internal/charmstore/migrations_test.go [hunk 2]
```diff
@@ -373,6 +378,11 @@ func (s *migrationsSuite) checkBaseEntitiesCount(c *gc.C, expectCount int) {
 func (s *migrationsSuite) insertEntity(c *gc.C, e *mongodoc.Entity, includeFields []string) {
 	c.Assert(includeFields, gc.Not(gc.HasLen), 0)
 
+	err := s.db.Entities().Insert(entityWithFields(c, e, includeFields))
+	c.Assert(err, gc.IsNil)
+}
+
+func entityWithFields(c *gc.C, e *mongodoc.Entity, includeFields []string) map[string]interface{} {
```
Comment (rogpeppe) [inline internal/charmstore/migrations_test.go:396]: doc comment.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1211/
Test PASSed.

File: internal/charmstore/migrations_test.go [hunk 3]
```diff
@@ -368,11 +382,18 @@ func (s *migrationsSuite) checkBaseEntitiesCount(c *gc.C, expectCount int) {
 	c.Assert(count, gc.Equals, expectCount)
 }
 
-// insertEntity inserts the given entity. Only the fields specified in includeFields
-// will be inserted.
-func (s *migrationsSuite) insertEntity(c *gc.C, e *mongodoc.Entity, includeFields []string) {
-	c.Assert(includeFields, gc.Not(gc.HasLen), 0)
+// insertEntity inserts the given entity. The migration that the entity
+// is to be inserted for is specified in name, only fields that existed
```
Comment (rogpeppe) [inline internal/charmstore/migrations_test.go]: s/,/;/

File: internal/charmstore/migrations_test.go [hunk 4]
```diff
@@ -341,12 +350,17 @@ func getMigrations(names ...mongodoc.MigrationName) (ms []migration) {
 	return ms
 }
 
-func (s *migrationsSuite) checkEntity(c *gc.C, expectEntity *mongodoc.Entity) {
+// checkEntity checks the entity stored in the database with the ID
+// expectEntity.URL is the same as expectEntity for all fields that exist
+// in the database following completion of the migration named name.
```
Comment (rogpeppe) [inline internal/charmstore/migrations_test.go]: s/of the migration named name/of the given migration/

?

File: internal/charmstore/migrations_test.go [hunk 5]
```diff
@@ -32,8 +32,13 @@ func (s *migrationsSuite) SetUpTest(c *gc.C) {
 	s.executed = nil
 }
 
+const (
+	beforeAllMigrations mongodoc.MigrationName = "---"
```
Comment (rogpeppe) [inline internal/charmstore/migrations_test.go]: or just "start" and "end" ?

File: internal/charmstore/migrations_test.go [hunk 6]
```diff
@@ -256,14 +265,14 @@ func (s *migrationsSuite) TestMigrateParallelMigration(c *gc.C) {
 		Size:           12,
 	}
 	denormalizeEntity(e1)
-	s.insertEntity(c, e1, initialEntityFields)
+	s.insertEntity(c, e1, beforeAllMigrations)
```
Comment (rogpeppe) [inline internal/charmstore/migrations_test.go:268]: This reads very nicely.

**Comment by rogpeppe** [comment]
LGTM, thanks.

File: internal/charmstore/migrations_test.go (hunk 5 above)
Comment (mhilton) [inline internal/charmstore/migrations_test.go]: done

File: internal/charmstore/migrations_test.go (hunk 6 above)
Comment (mhilton) [inline internal/charmstore/migrations_test.go:268]: thanks, I thought so

File: internal/charmstore/migrations_test.go (hunk 4 above)
Comment (mhilton) [inline internal/charmstore/migrations_test.go]: done

File: internal/charmstore/migrations_test.go (hunk 3 above)
Comment (mhilton) [inline internal/charmstore/migrations_test.go]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1224/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**